lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -4568,18 +4484,39 @@ void PatchInOutImportExport::createTessBufferStoreFunction() {
   Value *tfBufferOffset = BinaryOperator::CreateMul(relPatchId, tfByteStride, "", tfStoreBlock);
 
   tfBufferOffset = BinaryOperator::CreateAdd(tfBufferOffset, tfByteOffset, "", tfStoreBlock);
-  tfBufferOffset = BinaryOperator::CreateAdd(tfBufferOffset, tfBufferBase, "", tfStoreBlock);
 
   auto branch = BranchInst::Create(endBlock, tfStoreBlock);
 
+  // Create llvm.amdgcn.raw.tbuffer.store.
+  SmallVector<unsigned> formats;
+  if (m_gfxIp.major <= 9) {
+    formats = {
+        ((BUF_NUM_FORMAT_FLOAT << 4) | (BUF_DATA_FORMAT_32)),
+        ((BUF_NUM_FORMAT_FLOAT << 4) | (BUF_DATA_FORMAT_32_32)),
+        ((BUF_NUM_FORMAT_FLOAT << 4) | (BUF_DATA_FORMAT_32_32_32)),
+        ((BUF_NUM_FORMAT_FLOAT << 4) | (BUF_DATA_FORMAT_32_32_32_32)),
+    };
+  } else if (m_gfxIp.major == 10) {
+    formats = {BUF_FORMAT_32_FLOAT, BUF_FORMAT_32_32_FLOAT, BUF_FORMAT_32_32_32_FLOAT, BUF_FORMAT_32_32_32_32_FLOAT};
+  } else {
+    llvm_unreachable("Not implemented!");
+  }
+
+  Twine callName = Twine("llvm.amdgcn.raw.tbuffer.store.") + getTypeName(tfValueTy);
```

Comment:
you can't stack-allocate a Twine; it's safest to only use it as a temporary: https://llvm.org/doxygen/classllvm_1_1Twine.html#details

Reply:
Thanks. Fixed.

Reply:
Fixed.